Makefile

```
@@ -63,4 +63,23 @@ clean :
 	@ find . -iname '*.pyc' -delete
 	@ find . -iname '__pycache__' -delete
 
+# this target requires a little explanation... when the private-data-objects
+# submodule is updated, it will generally lack a branch name (git checkout
+# results in a detached HEAD state). in order to build from the submodule
+# we need the submodule to belong to a branch; in this case we are just going
+# to create a branch that is named by the current pdo-contracts version.
+# PDO_SOURCE_ROOT must be set to the submodule directory to build correctly.
+pdo_docker :
+	PDO_SOURCE_ROOT=$(SOURCE_ROOT)/private-data-objects \
+		cd $(SOURCE_ROOT)/private-data-objects && git checkout -B pdo-contracts-${VERSION}
+	PDO_SOURCE_ROOT=$(SOURCE_ROOT)/private-data-objects \
+		make -C private-data-objects/docker all
```

To make sure that after submodule update you get correct version ....
```suggestion
		make -C private-data-objects/docker clean all
```

---

why is the clean necessary? we'll rebuild the correct version even if images exist for other versions.

---

if in a pr you build, then pr pushes a submodule update, you dutifully try to rebuild after submodule update but without clean the repository dir in the submodule still will be the old one, so you build the wrong stuff ...